### PR TITLE
[JENKINS-25736] Ability to pause a running flow

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-25736](https://issues.jenkins-ci.org/browse/JENKINS-25736)

Originally filed by @kohsuke as https://github.com/jenkinsci/pipeline-plugin/pull/195.

- [X] basic implementation
- [X] make it survive a restart
- [X] also solve [JENKINS-32015](https://issues.jenkins-ci.org/browse/JENKINS-32015)
- [x] provide UI

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/cps/PauseUnpauseAction.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return null; // special presentation
+    }
+
+    @Override public String getDisplayName() {
+        return null; // special presentation
+    }
+
+    @RequirePOST
+    public void doToggle() {
+        execution.pause(!execution.isPaused());
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/PauseUnpauseAction.java]: No authentication check?  would expect at least build..

**Comment by jtnord** [comment]
what appears to be a :bug: as it looks to me like anyone can pause/unpause a pipeline.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/PauseUnpauseAction.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/PauseUnpauseAction.java]: Good point, I think this needs a permissions check, will add one.

**Comment by jtnord** [comment]
:bee:

**Comment by abayer** [comment]
:bee:

**Pull request merged**